lib/hpax/table.ex

```diff
@@ -264,24 +264,22 @@ defmodule HPAX.Table do
   In all cases, the table's `:protocol_max_table_size` is updated accordingly
   """
   @spec resize(t(), non_neg_integer()) :: t()
-  def resize(%__MODULE__{max_table_size: max_table_size} = table, new_protocol_max_table_size)
-      when new_protocol_max_table_size >= max_table_size do
-    %__MODULE__{
-      table
-      | protocol_max_table_size: new_protocol_max_table_size,
-        max_table_size: new_protocol_max_table_size
-    }
-  end
-
   def resize(%__MODULE__{} = table, new_protocol_max_table_size) do
     pending_minimum_resize =
       case table.pending_minimum_resize do
         nil -> new_protocol_max_table_size
         current -> min(current, new_protocol_max_table_size)
       end
 
+    table =
+      if new_protocol_max_table_size < table.size do
+        evict_to_size(table, new_protocol_max_table_size)
+      else
+        table
+      end
```

Comment:
This is making my brain hurt 😄 Here, we say to evict the table only if the protocol size goes *below* the table size. But the added clause to `evict_to_size/2` is a no-op if we try to evict to a *larger* size. These are opposite I think, so it might make sense, but I think some comments would be helpful?

Reply:
Yeah you're right, we don't need this one. I'd had these two commits originally staged in the opposite order so didn't notice that this chunk is now obsolete. Removing.